src/uct/tcp/tcp_listener.c

```diff
@@ -46,37 +40,36 @@ static void uct_tcp_listener_conn_req_handler(int fd, void *arg)
          * (so that send/recv won't be blocking) */
         status = ucs_sys_fcntl_modfl(accept_fd, O_NONBLOCK, 0);
         if (status != UCS_OK) {
-            close(accept_fd);
-            return;
-
+            goto err;
         }
 
-        listen_ctx = ucs_malloc(sizeof(uct_tcp_listener_ctx_t), "accept listen_ctx");
-        if (listen_ctx == NULL) {
+        sa_arg_ctx = ucs_malloc(sizeof(uct_tcp_sa_arg_t), "accept sa_arg_ctx");
+        if (sa_arg_ctx == NULL) {
             ucs_error("failed to allocate memory for a listen_ctx");
-            close(accept_fd);
-            return;
+            goto err;
         }
 
-        listen_ctx->fd  = accept_fd;
-        async_ctx       = listener->super.cm->iface.worker->async;
+        sa_arg_ctx->fd = accept_fd;
+        async_ctx      = listener->super.cm->iface.worker->async;
```

Comment:
is this not addressed or we really need to reinit `async_ctx` every iteration of the `for (;;)` loop?